gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystem.java

```diff
@@ -931,19 +931,50 @@ private void repairImplicitDirectory(Future<GoogleCloudStorageItemInfo> infoFutu
    * @param pageToken the page token to list
    */
   public ListPage<FileInfo> listFileInfoForPrefixPage(
-      URI prefix, ListFileOptions listOptions, String pageToken) throws IOException {
+          URI prefix, ListFileOptions listOptions, String pageToken) throws IOException {
     logger.atFiner().log(
-        "listAllFileInfoForPrefixPage(prefix: %s, pageToken:%s)", prefix, pageToken);
+            "listAllFileInfoForPrefixPage(prefix: %s, pageToken:%s)", prefix, pageToken);
+    return listFileInfoForPrefixPage(prefix, listOptions, pageToken, true);
+  }
+
+  /**
+   * Equivalent to {@link #listFileInfoForPrefixPage} but returns {@link FileInfo}s listed by single
+   * request (1 page).
+   *
+   * @param prefix    the prefix to use to list all matching objects.
+   * @param pageToken the page token to list
+   * @param recursive the flag to scan recursively
+   */
+  public ListPage<FileInfo> listFileInfoForPrefixPage(
+          URI prefix, String pageToken, boolean recursive) throws IOException {
```

Comment:
`listFileInfoForPrefixPage` method already [supports parametrization](https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/4b4d659b317e9018852cc43ba2b15930910e5d80/gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystem.java#L933-L934) via `ListFileOptions` class: https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/4b4d659b317e9018852cc43ba2b15930910e5d80/gcsio/src/main/java/com/google/cloud/hadoop/gcsio/ListFileOptions.java#L24

May you add `recursive` option to the `ListFileOptions` class instead?

Reply:
+1